src/oom/oomd-util.c

```diff
@@ -0,0 +1,352 @@
+                if (r < 0)
+                        return r;
+
+                log_info("oomd dry-run: Would have tried to kill %s with recurse=%s", cg_path, true_false(recurse));
+                return 0;
+        }
+
+        pids_killed = set_new(NULL);
+        if (!pids_killed)
+                return log_oom();
+
+        r = increment_oomd_xattr(path, OOMD_KILL_INIT_XATTR, 1);
+        if (r < 0)
+                return r;
+
+        if (recurse)
+                r = cg_kill_recursive(SYSTEMD_CGROUP_CONTROLLER, path, SIGKILL, CGROUP_IGNORE_SELF, pids_killed, log_kill, NULL);
+        else
+                r = cg_kill(SYSTEMD_CGROUP_CONTROLLER, path, SIGKILL, CGROUP_IGNORE_SELF, pids_killed, log_kill, NULL);
+        if (r < 0)
+                return r;
+
+        r = increment_oomd_xattr(path, OOMD_KILL_DONE_XATTR, set_size(pids_killed));
+        if (r < 0)
+                return r;
+
+        return 0;
+}
+
+int oomd_cgroup_context_get(const char *path, OomdCGroupContext *ret) {
+        _cleanup_free_ char *p = NULL, *val = NULL;
+        OomdCGroupContext ctx = {0};
+        int r;
+
+        assert(path);
+        assert(ret);
+
+        ctx.is_root = streq("/", empty_to_root(path));
+
+        r = cg_get_path(SYSTEMD_CGROUP_CONTROLLER, path, "memory.pressure", &p);
+        if (r < 0)
+                return r;
+
+        r = read_resource_pressure(p, PRESSURE_TYPE_FULL, &ctx.memory_pressure);
+        if (r < 0)
+                return r;
+
+        if (ctx.is_root) {
+                r = procfs_memory_get_used(&ctx.current_memory_usage);
+                if (r < 0)
+                        return r;
+        } else {
+                r = cg_get_attribute_as_uint64(SYSTEMD_CGROUP_CONTROLLER, path, "memory.current", &ctx.current_memory_usage);
+                if (r < 0)
+                        return r;
+
+                r = cg_get_attribute_as_uint64(SYSTEMD_CGROUP_CONTROLLER, path, "memory.min", &ctx.memory_min);
+                if (r < 0)
+                        return r;
+
+                r = cg_get_attribute_as_uint64(SYSTEMD_CGROUP_CONTROLLER, path, "memory.low", &ctx.memory_low);
+                if (r < 0)
+                        return r;
+
+                r = cg_get_attribute_as_uint64(SYSTEMD_CGROUP_CONTROLLER, path, "memory.swap.current", &ctx.swap_usage);
+                if (r < 0)
+                        return r;
+
+                r = cg_get_keyed_attribute(SYSTEMD_CGROUP_CONTROLLER, path, "memory.stat", STRV_MAKE("pgscan"), &val);
+                if (r < 0)
+                        return r;
+
+                r = safe_atou64(val, &ctx.pgscan);
+                if (r < 0)
+                        return r;
+        }
+
+        *ret = ctx;
+        return 0;
+}
+
+int oomd_system_context_get(const char *proc_swaps_path, OomdSystemContext *ret) {
+        _cleanup_fclose_ FILE *f = NULL;
+        OomdSystemContext ctx = {0};
```

Comment:
{} is sufficient too as initializer

(not strictly ansi, but totally fine on gnu/llvm and makes a lot more sense)